ksqldb-streams/src/test/java/io/confluent/ksql/execution/streams/SourceBuilderV1Test.java

```diff
@@ -0,0 +1,1161 @@
+
+    // When:
+    final GenericRow withTimestamp = transformer.transform(key, row);
+
+    // Then:
+    assertThat(withTimestamp, equalTo(GenericRow.genericRow(
+        "baz", 123, A_ROWTIME, A_ROWPARTITION, A_ROWOFFSET, A_KEY, A_WINDOW_START, A_WINDOW_END)));
+  }
+
+  @Test
+  public void shouldAddRowTimeAndTimeWindowedRowKeyColumnsToLegacyTable() {
+    // Given:
+    givenWindowedSourceTableV1();
+    final ValueTransformerWithKey<Windowed<GenericKey>, GenericRow, GenericRow> transformer =
+        getTransformerFromTableSource(windowedTableSourceV1);
+
+    final Windowed<GenericKey> key = new Windowed<>(
+        KEY,
+        new TimeWindow(A_WINDOW_START, A_WINDOW_END)
+    );
+
+    // When:
+    final GenericRow withTimestamp = transformer.transform(key, row);
+
+    // Then:
+    assertThat(withTimestamp,
+        is(GenericRow.genericRow("baz", 123, A_ROWTIME, A_KEY, A_WINDOW_START, A_WINDOW_END)));
+  }
+
+  @Test
+  public void shouldThrowOnTryToGetTransformerFromIncompatibleTableSource() {
+    // Given:
+    givenWindowedSourceTableV1(ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION);
+
+    // When:
+    final Exception e = assertThrows(
+        KsqlException.class,
+        () -> getTransformerFromTableSource(windowedTableSourceV1)
+    );
+
+    // Then:
+    assertThat(e.getMessage(), containsString(
+        "Should not be using old execution step version with new queries"));
+  }
+
+  @Test
+  public void shouldAddRowTimeAndSessionWindowedRowKeyColumnsToStreamLegacy() {
+    // Given:
+    givenWindowedSourceStream(LEGACY_PSEUDOCOLUMN_VERSION_NUMBER);
+    final ValueTransformerWithKey<Windowed<GenericKey>, GenericRow, GenericRow> transformer =
+        getTransformerFromStreamSource(windowedStreamSource);
+
+    final Windowed<GenericKey> key = new Windowed<>(
+        KEY,
+        new SessionWindow(A_WINDOW_START, A_WINDOW_END)
+    );
+
+    // When:
+    final GenericRow withTimestamp = transformer.transform(key, row);
+
+    // Then:
+    assertThat(withTimestamp,
+        equalTo(GenericRow.genericRow("baz", 123, A_ROWTIME, A_KEY, A_WINDOW_START, A_WINDOW_END)));
+  }
+
+  @Test
+  public void shouldAddPseudoColumnsAndSessionWindowedRowKeyColumnsToStream() {
+    // Given:
+    givenWindowedSourceStream();
+    final ValueTransformerWithKey<Windowed<GenericKey>, GenericRow, GenericRow> transformer =
+        getTransformerFromStreamSource(windowedStreamSource);
+
+    final Windowed<GenericKey> key = new Windowed<>(
+        KEY,
+        new SessionWindow(A_WINDOW_START, A_WINDOW_END)
+    );
+
+    // When:
+    final GenericRow withTimestamp = transformer.transform(key, row);
+
+    // Then:
+    assertThat(withTimestamp,
+        equalTo(GenericRow.genericRow(
+            "baz", 123, A_ROWTIME, A_ROWPARTITION, A_ROWOFFSET, A_KEY, A_WINDOW_START, A_WINDOW_END)));
+  }
+
+  @Test
+  public void shouldAddRowTimeAndSessionWindowedRowKeyColumnsToTableLegacy() {
+    // Given:
+    givenWindowedSourceTableV1();
```

Comment:
This needs to change to explicitly pass the legacy pseducolumn version, once the issue with `givenWindowedSourceTableV1()` (see previous comment) is addressed. It'd be good to audit the rest of the tests for these types of typos as well.

Reply:
The 0-arg overloads in `given{Windowed}SourceTableV1` were meant to use the `LEGACY_PSEUDOCOLUMN_VERSION_NUMBER`, as our default case is that they'll never be touched with a different pseudocolumn version. Is this a readability decision?

Reply:
Got it. I see the reasoning behind this but I think the readability cost makes it not worth it. It's really confusing that the default for `givenWindowedSourceStream()` is `ROWPARTITION_ROWOFFSET_PSEUDOCOLUMN_VERSION` but the default for `givenWindowedSourceTableV1()` is `LEGACY_PSEUDOCOLUMN_VERSION_NUMBER`.

How about we remove the overload for `givenWindowedSourceTableV1()` and always use the version that passes an explicit pseudocolumn version?

Reply:
Sounds good- I went ahead and did this for the unwindowed version too as I can't see why that'd be different.